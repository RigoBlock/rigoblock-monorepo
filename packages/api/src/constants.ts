export enum NETWORKS {
  MAINNET = '1',
  ROPSTEN = '3',
  KOVAN = '42',
  LOCALHOST = '5777'
}

export const RPC_URLS = {
  [NETWORKS.MAINNET]: 'wss://mainnet.infura.io/ws/',
  [NETWORKS.ROPSTEN]: 'wss://ropsten.infura.io/ws/',
  [NETWORKS.KOVAN]: 'wss://kovan.infura.io/ws/',
  [NETWORKS.LOCALHOST]: 'ws://localhost:8545'
}
