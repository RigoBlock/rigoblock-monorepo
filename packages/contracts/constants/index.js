const path = require('path')

const NETWORKS = {
  ganache: 'http://localhost:8545',
  mainnet: 'https://mainnet.infura.io/v3/d48872aa1c00471c825e9d856c3c3138',
  ropsten: 'https://ropsten.infura.io/v3/d48872aa1c00471c825e9d856c3c3138',
  kovan: 'https://kovan.infura.io/v3/d48872aa1c00471c825e9d856c3c3138'
}

module.exports = {
  TMP_DIR: '.tmp',
  ARTIFACTS_DIR: path.resolve(__dirname, '../artifacts'),
  CONTRACTS_DIR: path.resolve(__dirname, '../src'),
  NETWORKS,
  GANACHE_PORT: 8545,
  GANACHE_NETWORK_ID: 5777,
  GAS_ESTIMATE: 5e6,
  CONTRACT_NAMES: [
    'AEthfinex',
    'ATotlePrimary',
    'AWeth',
    'Authority',
    'Distribution',
    'Drago',
    'DragoEventful',
    'DragoFactory',
    'DragoRegistry',
    'ERC20',
    'Exchange',
    'ExchangeEfx',
    'ExchangeV1Fork',
    'ExchangesAuthority',
    'Faucet',
    'HGetDragoData',
    'Inflation',
    'Migrations',
    'NavVerifier',
    'Owned',
    'OwnedUninitialized',
    'ProofOfPerformance',
    'RigoToken',
    'SigVerifier',
    'TokenTransferProxy',
    'TotlePrimary',
    'UnlimitedAllowanceToken',
    'Vault',
    'VaultEventful',
    'VaultFactory',
    'WETH9',
    'WrapperLock',
    'WrapperLockEth',
    'ZeroExExchangeHandler'
  ]
}
