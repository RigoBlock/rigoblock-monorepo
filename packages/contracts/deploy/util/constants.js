export const constants = {
    optimizerEnabled: false,
    timeoutMs: 30000,
    contracts: '*' as '*'
}
