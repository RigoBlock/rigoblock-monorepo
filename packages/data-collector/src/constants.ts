export const REDIS_HOST = process.env.REDIS_HOST || '127.0.0.1'
export const REDIS_PORT = parseInt(process.env.REDIS_PORT) || 6379
export const REDIS_PASSWORD = process.env.REDIS_PASSWORD || ''
export const TELEGRAF_HOST = process.env.TELEGRAF_HOST || '127.0.0.1'
export const TELEGRAF_PORT = process.env.TELEGRAF_PORT || 8125

export const TOKEN_MARKET_BASE_URL =
  'https://tokenmarket.net/blockchain/ethereum/assets'
export const CRYPTO_NEWS_BASE_URL =
  'https://cryptopanic.com/api/posts/?auth_token=5b3f861c261e7e6243a55dc7b7ca600832156886&public=true&filter=hot&currencies='
