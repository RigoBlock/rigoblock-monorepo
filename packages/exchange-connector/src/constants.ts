export const MAINNET: string = 'mainnet'
export const ROPSTEN: string = 'ropsten'
export const KOVAN: string = 'kovan'
export const RINKEBY: string = 'rinkeby'

export const PRICE_PRECISION: number = 7
export const AMOUNT_PRECISION: number = 5
export const TO_WEI: number = 1e18

export enum supportedExchanges {
  ETHFINEX = 'Ethfinex',
  ETHFINEX_RAW = 'EthfinexRaw',
  ERCDEX = 'ERCdEX',
  ZEROEXRELAYER = '0xStandardRelayer'
}

export enum NETWORKS {
  MAINNET = 1,
  KOVAN = 42,
  ROPSTEN = 3,
  RINKEBY = 4
}

export enum WS_STATUS {
  CLOSED = 'closed',
  OPEN = 'open',
  CONNECTING = 'connecting',
  CLOSING = 'closing'
}
