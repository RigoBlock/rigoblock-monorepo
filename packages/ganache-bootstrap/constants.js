module.exports = {
  GANACHE_URL: 'http://localhost:8545'
}
