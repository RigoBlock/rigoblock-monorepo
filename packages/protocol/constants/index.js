const path = require('path')

const NETWORKS = process.env.NETWORKS
  ? process.env.NETWORKS.split(',')
  : ['http://localhost:8545']

module.exports = {
  ARTIFACTS_DIR: path.resolve(__dirname, '../artifacts'),
  CONTRACTS_DIR: path.resolve(__dirname, '../solidity-contracts'),
  NETWORKS,
  GANACHE_PORT: 8545,
  GANACHE_NETWORK_ID: 5777,
  GAS_ESTIMATE: 5e6,
  CONTRACT_NAMES: [
    'Authority',
    'Distribution',
    'Drago',
    'DragoEventful',
    'DragoFactory',
    'DragoRegistry',
    'ERC20',
    'Inflation',
    'Migrations',
    'Owned',
    'OwnedUninitialized',
    'ProofOfPerformance',
    'RigoToken',
    'TokenTransferProxy',
    'UnlimitedAllowanceToken',
    'Vault',
    'VaultEventful',
    'VaultFactory',
    'WrapperLock',
    'WrapperLockEth'
  ]
}
