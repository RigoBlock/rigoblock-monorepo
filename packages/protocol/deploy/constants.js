const path = require('path')

module.exports = {
  ARTIFACTS_DIR: path.resolve(__dirname, '../artifacts'),
  CONTRACTS_DIR: path.resolve(__dirname, '../contracts'),
  NETWORKS: process.env.NETWORKS || ['http://localhost:8545'],
  GAS_ESTIMATE: 5e6
}
