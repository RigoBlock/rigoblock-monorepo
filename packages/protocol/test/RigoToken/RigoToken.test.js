import { BigNumber } from 'bignumber.js'
import { GAS_ESTIMATE } from '../../constants'

const contractName = 'RigoToken'

describe(contractName, () => {
  const RIGOTOKEN = 'Rigo Token'
  const RIGOTOKEN_SYMBOL = 'GRG'
  const RIGOTOKEN_DECIMALS = 18
  describe('getName', () => {
    it('returns the contract name', async () => {
      const name = await baseContracts[contractName].getName()
      expect(name).toBe(RIGOTOKEN)
    })
  })
  describe('getSymbol', () => {
    it('returns the contract symbol', async () => {
      const symbol = await baseContracts[contractName].getSymbol()
      expect(symbol).toBe(RIGOTOKEN_SYMBOL)
    })
  })
  describe('getDecimals', () => {
    it("returns Rigo Token's decimals", async () => {
      const decimals = await baseContracts[contractName].getDecimals()
      expect(decimals).toEqual(new BigNumber(RIGOTOKEN_DECIMALS))
    })
  })
  describe('getMinter', () => {
    it('returns the address of the minter', async () => {
      const minter = await baseContracts[contractName].getMinter()
      expect(minter).toBe(accounts[0])
    })
  })
  describe('getRigoblock', () => {
    it('returns the address of Rigoblock DAO', async () => {
      const dao = await baseContracts[contractName].getRigoblock()
      expect(dao).toBe(accounts[0])
    })
  })
  // describe('getInflationFactor', () => {
  //   it('returns something', async () => {
  //     const inflactionFactor = await baseContracts[
  //       contractName
  //     ].getInflationFactor(baseContracts['VaultFactory'].address)
  //     console.log(inflactionFactor)
  //   })
  // })
  describe('mintToken', () => {
    it('mints new tokens and emits a TokenMinted event', async () => {
      const tokenAmount = 10
      const txHash = await baseContracts[contractName].mintToken(
        accounts[0],
        tokenAmount
      )
      const mintEvent = baseContracts[contractName].TokenMinted()
      const eventsPromise = new Promise((resolve, reject) => {
        mintEvent.get(
          (err, data) => (err ? reject(new Error(err)) : resolve(data))
        )
      })
      const events = await eventsPromise
      expect(txHash).toBeHash()
      expect(events.pop().args).toEqual({
        recipient: accounts[0],
        amount: new BigNumber(tokenAmount)
      })
    })
    it('can only be called by the minter', async () => {
      const tokenAmount = 10
      await expect(
        baseContracts[contractName].mintToken.sendTransactionAsync(
          accounts[1],
          tokenAmount,
          {
            // non minter account
            from: accounts[1],
            gas: GAS_ESTIMATE,
            gasPrice: 1
          }
        )
      ).rejects.toThrowErrorMatchingSnapshot()
    })
  })
  describe('changeMintingAddress', () => {
    it('sets a new minter given a valid account address', async () => {
      const txHash = await baseContracts[contractName].changeMintingAddress(
        accounts[1]
      )
      expect(txHash).toBeHash()
      const newMinter = await baseContracts[contractName].getMinter()
      expect(newMinter).toBe(accounts[1])
    })
    it('can only be called by the rigoblock DAO', async () => {
      await expect(
        baseContracts[contractName].changeMintingAddress.sendTransactionAsync(
          accounts[1],
          {
            // non DAO account
            from: accounts[1],
            gas: GAS_ESTIMATE,
            gasPrice: 1
          }
        )
      ).rejects.toThrowErrorMatchingSnapshot()
    })
  })
  describe('changeRigoblockAddress', () => {
    it('sets a new dao', async () => {
      const txHash = await baseContracts[contractName].changeRigoblockAddress(
        accounts[1]
      )
      expect(txHash).toBeHash()
      const newDao = await baseContracts[contractName].getRigoblock()
      expect(newDao).toBe(accounts[1])
    })
    it('can only be called by the rigoblock DAO', async () => {
      await expect(
        baseContracts[contractName].changeRigoblockAddress.sendTransactionAsync(
          accounts[1],
          {
            // non DAO account, we previously changed it
            from: accounts[0],
            gas: GAS_ESTIMATE,
            gasPrice: 1
          }
        )
      ).rejects.toThrowErrorMatchingSnapshot()
    })
  })
})
