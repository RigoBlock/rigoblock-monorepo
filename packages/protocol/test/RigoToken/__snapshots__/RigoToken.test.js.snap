// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RigoToken changeMintingAddress can only be called by the rigoblock DAO 1`] = `"VM Exception while processing transaction: revert"`;

exports[`RigoToken changeRigoblockAddress can only be called by the rigoblock DAO 1`] = `"VM Exception while processing transaction: revert"`;

exports[`RigoToken mintToken can only be called by the minter 1`] = `"VM Exception while processing transaction: revert"`;
